### ISSUE TITLE
runtests.py no longer builds on windows

### ISSUE THREAD

**Issue by eric-wieser** [numpy/numpy#13689]
Building on a commit derived from 849e4d429951ad93691e6149d070b9d31536cc12:

```text
PS C:\Users\wiese\Repos\numeric-python\numpy> .\runtests.py
Building, see build.log...

Error compiling Cython file:
------------------------------------------------------------
...
                          int8_t, int16_t, int32_t, int64_t, intptr_t)
import numpy as np
cimport numpy as np
ctypedef np.npy_bool bool_t

from .common cimport bitgen_t
^
------------------------------------------------------------

bounded_integers.pxd:7:0: 'common.pxd' not found
Processing numpy/random\bounded_integers.pyx
Traceback (most recent call last):
  File "C:\Users\wiese\Repos\numeric-python\numpy\tools\cythonize.py", line 224, in <module>
    main()
  File "C:\Users\wiese\Repos\numeric-python\numpy\tools\cythonize.py", line 220, in main
    find_process_files(root_dir)
  File "C:\Users\wiese\Repos\numeric-python\numpy\tools\cythonize.py", line 212, in find_process_files
    process(cur_dir, fromfile, tofile, function, hash_db)
  File "C:\Users\wiese\Repos\numeric-python\numpy\tools\cythonize.py", line 171, in process
    processor_function(fromfile, tofile)
  File "C:\Users\wiese\Repos\numeric-python\numpy\tools\cythonize.py", line 86, in process_pyx
    [sys.executable, '-m', 'cython'] + flags + ["-o", tofile, fromfile])
  File "C:\Program Files\Python 3.5\lib\subprocess.py", line 271, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['C:\\Program Files\\Python 3.5\\python.exe', '-m', 'cython', '--fast-fail', '-o', 'bounded_integers.c', 'bounded_integers.pyx']' returned non-zero exit status 1
Running from numpy source directory.
Cythonizing sources
Traceback (most recent call last):
  File "setup.py", line 442, in <module>
    setup_package()
  File "setup.py", line 425, in setup_package
    generate_cython()
  File "setup.py", line 233, in generate_cython
    raise RuntimeError("Running cythonize failed!")
RuntimeError: Running cythonize failed!

Build failed!
```

**Comment by eric-wieser** [comment]
Caused by a weird mixed-state build, so not a regression. It seems that non-committed sources are cythonized automatically, even if they're stale left from a bad git reset and not referenced by a setup.py.

**Comment by charris** [comment]
Yep, that happens, feature or bug? I tend to the former, although a bad git reset isn't intentional.

**Comment by eric-wieser** [comment]
Inconsistent with how we handle new C files, which are only built if added to `setup.py`. Probably not worth worrying about this though, I think we can close it.

**Comment by charris** [comment]
OK, done :)

**Comment by tallamjr** [comment]
I also have this exact error output when attempting to install `numpy` from source. This was encountered at version: 
`1.17.0.dev0+7dc91de` and  `1.16.4`

Is there a way to suppress this or a workaround?

_Other versioning information_
- macOS [`10.14.5`]
- Python [`3.7.3`]
- Cython [`0.29.10`]

*DISCLAIMER:* I am very new to the `numpy` source code so unsure of the internals, I am hoping to contribute for the first time to this codebase

**Comment by mattip** [comment]
You are working from a git checkout (`git clone ...`) or from a download-and-untar of the sources? If the latter, where did you get the tarball?

**Comment by tallamjr** [comment]
`git clone` ..

I first forked `numpy` and then added the remote upstream such that `git remote -v` reveals:
```
origin	git@github.com:tallamjr/numpy.git (fetch)
origin	git@github.com:tallamjr/numpy.git (push)
upstream	git@github.com:numpy/numpy.git (fetch)
upstream	git@github.com:numpy/numpy.git (push)
```

Then then I tried to run `pip install .` in the main directory (following instructions relating to this issue https://github.com/numpy/numpy/issues/13114#issue-420711756)

To test version `v1.16.4` I just checkout that tag with `git checkout v1.6.4` and ran `pip install .` again, which produced the same error.

**Comment by mattip** [comment]
what version of cython are you using? You also should do `git clean -xfd` between build attempts, which will clean out all build artifacts

**Comment by tallamjr** [comment]
Running `git clean -xfd` allowed me to build fine on `v1.16.4` and `1.17.0.dev0+7dc91de`, thanks! 😃 

What I think happened is when I first cloned the repo and tried to build, it was not in a contained environment and did not have `cython` installed. Then when I tried again inside a conda environment with Cython I was getting the same error because I had not run `git clean -xfd`

Thanks again @mattip  for you assistance!
